back-end/API/app/middleware/numberVerify.py

```
@@ -0,0 +1,9 @@
+from app.customexception.CustomException import ValidationException
+
+def verifyNumber(field, data, formats):
+    for format in formats:
+        try:
+            if isinstance(format(data), format):
+                return True
+        except ValueError:    
+            raise ValidationException(field, "Ce champ doit correspondre a un nombre.")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)